packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -338,27 +339,29 @@ export class RunningSummarizer implements IDisposable {
                 { refreshLatestAck: true, fullTree: true, delaySeconds: 10 * 60 },
             ];
             let overrideDelaySeconds: number | undefined;
-            let totalAttempts = 0;
-            let attemptPerPhase = 0;
+            let summaryAttempts = 0;
+            let summaryAttemptsPerPhase = 0;
 
             let lastResult: { message: string; error: any; } | undefined;
 
-            for (let attemptPhase = 0; attemptPhase < attempts.length;) {
+            for (let summaryAttemptPhase = 0; summaryAttemptPhase < attempts.length;) {
                 if (this.cancellationToken.cancelled) {
                     return;
                 }
 
-                totalAttempts++;
-                attemptPerPhase++;
+                this.totalAttempts++;
```

Comment:
Unfortunately, no matter how we name these things, it will be confusing, and one would need to look at the code to understand what it tracks. So, I'm not sure my suggestion is helping, but....
I'd probably rename totalAttempts and would increment it outside of the loop only, to track successful attempts. I.e. I'd keep it = 1 for first 3 unsuccessful and first successful retries, and then move to 2. That way it tracks succeeded cases (I can tell how many summaries we actually produced per session), and I can drill into summaryAttempts if I want to see how many attempts it takes per successful summary.

Reply:
I see. So something like "RuningSummarizer.totalSuccessfulAttempts."